Complete step2-2

seongheum-choi opened the pull request:
Hyodol & Nailer
step 2-2 필수미션입니다.

seongheum-choi committed:
Complete step2-2

crongro commented on main.js:
```diff
@@ -1,49 +1,24 @@
-}
+(async () => {
+  try{
+    await polygon.getCircle(3)
```
여러개를 다 호출시키려니 await가 많아졌군요. ㅎ
전체적으로  async/await 를 써서 좀 헷갈릴 수 있었을 거 같은데, 비동기를 잘 다루셨어요.

crongro commented on polygon.js:
```diff
@@ -1,39 +1,138 @@
-    return "Type error"
+module.exports.getTrapezoid = async (height, underWidth, upperWidth) => {
+  var props = [height, underWidth, upperWidth]
+  if(!checkNumbers(props)){
```
props대신, arguments를 써도 될 거 같네요.

crongro commented on polygon.js:
```diff
@@ -1,39 +1,138 @@
+  var props = [height, underWidth, upperWidth]
+  if(!checkNumbers(props)){
+    var height = await input('[ trapezoid height ] >> ')
+    var underWidth = await input('[ trapezoid underWidth ] >> ')
```
이제 var를 쓰지 말도록 하세요~!
const/let 중 적당한 것을 쓰기.

crongro commented on polygon.js:
```diff
@@ -1,39 +1,138 @@
+  const trapezoid = height*(underWidth + upperWidth)/2
+  console.log('result trapezoid => ', trapezoid)
+  polygonCalls.push('trapezoid')
+  return true
```
반환값을 쓰는 곳이 없다면,  true가 없어도 될 듯.

crongro commented on polygon.js:
```diff
@@ -1,39 +1,138 @@
-  return r*r*height*PI;
+  var cylinder = radius*radius*height*PI
+  console.log('result cylinder => ', cylinder)
+  polygonCalls.push('cylinder')
```
이렇게 해도 되고요, 
함수를 조금더 활용해보겠다 싶으면, 이런것도 가능하겠죠.

```javascript
saveFunctionCallName('cylinder') 
```

crongro commented on polygon.js:
```diff
@@ -1,39 +1,138 @@
+
+module.exports.printExecutionSequence = () => {
+  var result = '계산 수행 순서 : '
+  polygonCalls.forEach(element => result = result.concat(element + ', '))
```
잘했어요.
여기 코드를 이렇게 개선해보세요.

- template literal로 문자열연산하기
- forEach  말고 reduce사용해보기.

seongheum-choi replied:
네 알겠습니다! 수정해서 다시 PR하겠습니다~!

crongro reviewed (changes requested):
javascript비동기를 잘 이해하고 있습니다.
💯 

몇개 남겨드린 리뷰로 개선해서 다시 push해주세요.

seongheum-choi committed:
Implement requested fixes

1. var를 const와 let으로 대체
2. 불필요한 true return 삭제
3. saveFunctoinCallName function 만들고 사용
4. template literal 사용, 문자열 연산 넣어둠
5. reduce method 사용

crongro reviewed (changes requested):
circle 를 재귀적으로 호출한 후에는,  전체 누적된 합을 구하는 게 문제에요. (각각의 결과를 출력하는 게 아니고)
요것도 좀 수정해보세요.

seongheum-choi committed:
Implement recursive function

원의 넓이들을 출력하지 않고 합을 출력합니다.

seongheum-choi commented:
재귀적으로 원의 넓이들을 더하고 출력하도록 수정하였습니다!

crongro reviewed (approved):
수고했어요.

추가로, 
재귀호출시 전역변수 없이 구현하는 것을 생각해보세요.

추가미션도 gogo!
